<launch>
    <!-- this program perform data format transformation on the solving results (the output workspace) -->
    <!--
        supported data format for transformation:
        0. JSON
        1. XML
        2. YAML
        3. BINARY (not recommended)
    -->
    <node pkg="ikalibr" type="ikalibr_data_format_transformer"
        name="ikalibr_data_format_transformer"
        output="screen">
        <!-- this pre path would be added to the head of each elem in {ws_dir_vec} -->
        <param name="pre_dir" value="$(find ikalibr)" type="string" />
        <!-- from 'src_format' to 'dst_format' -->
        <!--  
            supported data output format:
            0. JSON
            1. XML
            2. YAML
            3. BINARY (not recommended)
        -->
        <!-- data format transformation example from YAML (2) to JSON (0) -->
        <param name="src_format" value="YAML" type="string" />
        <param name="dst_format" value="JSON" type="string" />

        <!-- the ${ws_dir_vec} would be appended after the ${pre_dir} -->
        <!-- multiple workspaces are supported, you just need to add them in the below list -->
        <rosparam param="ws_dir_vec">
            [
            "data/real-world-old/data_202392616822"
            ]
        </rosparam>
    </node>

    <!--
         iKalibr: Unified Targetless Spatiotemporal Calibration Framework
         Copyright 2024, the School of Geodesy and Geomatics (SGG), Wuhan University, China
         https://github.com/Unsigned-Long/iKalibr.git

         Author: Shuolong Chen (shlchen@whu.edu.cn)
         GitHub: https://github.com/Unsigned-Long
          ORCID: 0000-0002-5283-9057

         Purpose: See .h/.hpp file.

         Redistribution and use in source and binary forms, with or without
         modification, are permitted provided that the following conditions are met:

         * Redistributions of source code must retain the above copyright notice,
           this list of conditions and the following disclaimer.
         * Redistributions in binary form must reproduce the above copyright notice,
           this list of conditions and the following disclaimer in the documentation
           and/or other materials provided with the distribution.
         * The names of its contributors can not be
           used to endorse or promote products derived from this software without
           specific prior written permission.

         THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
         AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
         IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
         ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
         LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
         CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
         SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
         INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
         CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
         ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
         POSSIBILITY OF SUCH DAMAGE.
    -->
</launch>